File: src/main/scala/es/juanc/katas/bankkata/Account.scala
package es.juanc.katas.bankkata

case class Account(transactionRepository: TransactionRepository, statementTransactions: StatementTransactions) {

	def deposit(amount: Int): Unit = transactionRepository.addTransactionWith(amount)

	def withdraw(amount: Int): Unit = transactionRepository.addTransactionWith(-amount)

	def printStatement(): Unit = statementTransactions.print {
			transactionRepository.allTransactions
		}
}

File: javascript/lib/Account.js
function Account ({transactionRepository, statementPrinter}) {

	return {

		deposit (amount = 0) {
			transactionRepository.addTransactionWith(amount)
		},

		withdrawal (amount = 0) {
			transactionRepository.addTransactionWith(-amount)
		},

		printStatement () {
			statementPrinter.print(transactionRepository.allTransactions())
		}
	}
}

module.exports = Account
